autodispose/src/main/java/com/uber/autodispose/ParallelFlowableScoper.java

```diff
@@ -0,0 +1,52 @@
+package com.uber.autodispose;
+
+import org.reactivestreams.Subscriber;
+
+import io.reactivex.Maybe;
+import io.reactivex.parallel.ParallelFlowable;
+import io.reactivex.parallel.ParallelFlowableConverter;
+
+class ParallelFlowableScoper<T> extends Scoper
+    implements ParallelFlowableConverter<T, ParallelFlowableSubscribeProxy<T>> {
+
+  ParallelFlowableScoper(Maybe<?> scope) {
+    super(scope);
+  }
+
+  @Override public ParallelFlowableSubscribeProxy<T> apply(final ParallelFlowable<T> upstream) {
+    return new ParallelFlowableSubscribeProxy<T>() {
+      @Override public void subscribe(Subscriber<? super T>[] subscribers) {
+        new AutoDisposeParallelFlowable<>(upstream, scope()).subscribe(subscribers);
+      }
+    };
+  }
+
+  static final class AutoDisposeParallelFlowable<T> extends ParallelFlowable<T> {
+
+    private final ParallelFlowable<T> source;
+    private final Maybe<?> scope;
+
+    AutoDisposeParallelFlowable(ParallelFlowable<T> source, Maybe<?> scope) {
+      this.source = source;
+      this.scope = scope;
+    }
+
+    @Override public void subscribe(Subscriber<? super T>[] subscribers) {
+      if (!validate(subscribers)) {
+        return;
+      }
+
+      Subscriber<? super T>[] newSubscribers = new Subscriber[subscribers.length];
+      for (int i = 0; i < subscribers.length; i++) {
+        AutoDisposingSubscriberImpl<? super T> subscriber =
```

Comment:
I kind of wonder if we do need a separate subscriber that manages all the subscribers here. @akarnokd penny for your thoughts?

Reply:
Parallel rails are largely independent and act as their own Flowables. That's why many parallel operators just delegate to their sequential versions.

Reply:
cool, in that case we can leave this as-is :+1: